src/main/scala/se/nimsa/sbx/box/BoxPushActor.scala

```diff
@@ -16,189 +16,127 @@
-    self ! PollOutgoing
-  }
-
-  override def postStop() =
-    poller.cancel()
-
-  val pool = Http().superPool[String]()
-
-  override def callAnonymizationService[R: ClassTag](message: Any) = anonymizationService.ask(message).mapTo[R]
-  override def callMetaDataService[R: ClassTag](message: Any) = metaDataService.ask(message).mapTo[R]
-  override def scheduleTask(delay: FiniteDuration)(task: => Unit) = system.scheduler.scheduleOnce(delay)(task)
-
-  def receive = LoggingReceive {
-    case PollOutgoing =>
-      context.become(inTransferState)
-      processNextOutgoingTransaction().onComplete {
-        case Success(_) =>
-          self ! TransferFinished
-          self ! PollOutgoing
-        case Failure(e) => e match {
-          case _@(EmptyTransactionException | RemoteBoxUnavailableException) =>
-            self ! TransferFinished
-          case e: Exception =>
-            SbxLog.error("Box", s"Failed to send file to box ${box.name}: ${e.getMessage}")
-            self ! TransferFinished
-        }
-      }
-  }
+  override implicit val system: ActorSystem = context.system
+  override implicit val ec: ExecutionContext = context.dispatcher
+  override implicit val scheduler: Scheduler = system.scheduler
 
-  def inTransferState: Receive = {
-    case TransferFinished => context.unbecome()
-  }
+  override protected val streamChunkSize: Long = storage.streamChunkSize
 
-  def processNextOutgoingTransaction(): Future[Unit] = {
-    boxService.ask(GetNextOutgoingTransactionImage(box.id)).mapTo[Option[OutgoingTransactionImage]].flatMap {
-      case Some(transactionImage) => sendFileForOutgoingTransactionImage(transactionImage)
-      case None => Future.failed(EmptyTransactionException)
-    }
-  }
+  val http = Http(system)
 
-  def sendFileForOutgoingTransactionImage(transactionImage: OutgoingTransactionImage): Future[Unit] =
-    boxService.ask(GetOutgoingTagValues(transactionImage)).mapTo[Seq[OutgoingTagValue]]
-      .flatMap(transactionTagValues =>
-        sendFile(transactionImage, transactionTagValues))
-
-  def sendFile(transactionImage: OutgoingTransactionImage, tagValues: Seq[OutgoingTagValue]): Future[Unit] = {
-    log.debug(s"Pushing transaction image $transactionImage with tag values $tagValues")
-
-    pushImagePipeline(transactionImage, tagValues).transformWith {
-      case Success(response) if response.status.intValue >= 200 && response.status.intValue < 300 =>
-        handleFileSentForOutgoingTransaction(transactionImage)
-      case Success(response) =>
-        Unmarshal(response).to[String].flatMap { errorMessage =>
-          handleFileSendFailedForOutgoingTransaction(transactionImage, response.status.intValue, new Exception(s"File send failed with status code ${response.status.intValue}: $errorMessage"))
+  override protected val pool: Flow[(HttpRequest, OutgoingTransactionImage), (Try[HttpResponse], OutgoingTransactionImage), Http.HostConnectionPool] =
```

Comment:
The connection pool needs separate arguments for host and port and has separate constructors for http and https, so unfortunately we have to reverse engineer this info from the box urls.